### PR TITLE
Change from source to client

### PR DESCRIPTION
**Pull request by hanh090**
Change from source to clientOffset so that dragging component is exactly same in pointer coordinate

### CODE REVIEW

**Comment by LouisBrunner** [comment]
Hey @hanh090! Thanks a lot for your contribution!

This is definitively an improvement over the previous behavior (as seen in #5).
My only issue with it is that the cursor is now always grabbing the top left corner of the component, which looks even more odd when dragging non-rect shapes. Any chance you could look into getting it to grab in the center of the component or in the top-left slightly offset (halfway from the center)?

**Comment by hanh090** [comment]
yes, I can take a look on that, but it should in another PR?

**Comment by LouisBrunner** [comment]
I would prefer if it was the same, I am not sure I would like to publish it as-is

**Comment by hanh090** [comment]
hi @LouisBrunner, i was fix position problem, but it was failed in ci checking, I was investigated but no luck. can you help me?

File: packages/react-dnd-preview/src/usePreview.js [hunk 1]
```diff
@@ -2,28 +2,48 @@ import { useDragLayer } from 'react-dnd';
...
 
 const usePreview = () => {
   const collectedProps = useDragLayer((monitor) => {
     return {
-      currentOffset: monitor.getSourceClientOffset(),
+      currentOffset: calculatePointerPosition(monitor),
       isDragging: monitor.isDragging(),
       itemType: monitor.getItemType(),
       item: monitor.getItem(),
+      monitor,
```
Comment (hanh090) [inline packages/react-dnd-preview/src/usePreview.js:33]: export monitor to handle these case preview change the shape when dragging, or just get more information when needed

**Comment by LouisBrunner** [comment]
@hanh090 Sorry for the delay!

Unfortunately, I am still seeing weird behavior when the scale of the preview is different. However I will be merging your PR as-is if it fixes your issue.

I will be reworking the `usePreview_spec` so that your code can be fully tested (gotta get that 100% coverage!) and also adding a special example pages to test different cases of preview not being aligned. This should allow us to test and showcase #5 easily.

Once the new example is ready, I would love if you could do another PR to add an example of what situation you were running into which required this change!

**Comment by hanh090** [comment]
when you done, please inform me so that I can prepare example for that

**Comment by hanh090** [comment]
hi @LouisBrunner, seems like you forgot to bump version? I still see it stay at '6.0.2' in 'react-dnd-preview'

**Comment by LouisBrunner** [comment]
I haven't published it yet no. I am planning to finish these extra changes this week and then do a release.

**Comment by LouisBrunner** [comment]
Hi @hanh090,

Sorry for the delay, I have now merged my changes and the new examples (https://github.com/LouisBrunner/dnd-multi-backend/blob/master/packages/react-dnd-preview/examples/offset/App.js), if you could add a new `<Draggable />` with the custom styling that caused the preview to be offset, it might help make the system more future-proof.

**Pull request merged**